src/frontend/src/expr/mod.rs

```diff
@@ -288,20 +288,32 @@ impl ExprImpl {
             }
 
             fn visit_subquery(&mut self, subquery: &Subquery) -> bool {
-                use crate::binder::BoundSetExpr;
+                self.depth += 1;
+                let has = self.visit_bound_set_expr(&subquery.query.body);
+                self.depth -= 1;
+
+                has
+            }
+        }
 
+        impl Has {
+            fn visit_bound_set_expr(&mut self, set_expr: &BoundSetExpr) -> bool {
                 let mut has = false;
-                self.depth += 1;
-                match &subquery.query.body {
+                match set_expr {
                     BoundSetExpr::Select(select) => {
                         select.exprs().for_each(|expr| has |= self.visit_expr(expr))
                     }
                     BoundSetExpr::Values(values) => {
                         values.exprs().for_each(|expr| has |= self.visit_expr(expr))
                     }
-                }
-                self.depth -= 1;
-
+                    BoundSetExpr::Query(query) => {
+                        has = self.visit_bound_set_expr(&query.body);
```

Comment:
Since we called `bind_query` for this case, the expressions inside should also have `depth+1`.